showdown/battle.py

```diff
@@ -350,7 +350,8 @@ def from_json(self, user_json, first_turn=False):
 
             nickname = pkmn_dict[constants.IDENT]
             pkmn = Pokemon.from_switch_string(pkmn_dict[constants.DETAILS], nickname=nickname)
-            pkmn.ability = pkmn_dict[constants.REQUEST_DICT_ABILITY]
+            try: pkmn.ability = pkmn_dict[constants.REQUEST_DICT_ABILITY]
+            except: pkmn.ability = "noability"
```

Comment:
Couldn't this just be:

```suggestion
            pkmn.ability = pkmn_dict.get(constants.REQUEST_DICT_ABILITY, None)
```